In [2]:
import math
import collections

import pandas as pd

from keras import regularizers
from keras.models import Model, Sequential
from keras.layers import *
from keras.wrappers.scikit_learn import KerasClassifier

import params
from utils.sequence_data import data_to_sequences_and_labels
from utils.metrics import print_report_for_binary_classfier
from utils.preprocessing import probs_to_binary_classes
from utils.hyperparams import hyperparams_search
from utils.plot import plot_train_validation_metric

Using TensorFlow backend.


In [3]:
dataset = pd.read_csv("../datasets/data_for_models/dataset_1996-01-01_2019-08-22.csv",
                           header=0, parse_dates=[0], index_col=0)

INPUT_SHAPE = (params.LOOKBACK//params.STEP, dataset.shape[-1] )

In [4]:
#params for generating sequences 
train_max_idx = math.ceil(len(dataset)*params.TRAIN_RATIO)
val_max_idx = math.ceil(len(dataset)*(params.TRAIN_RATIO+params.VAL_RATIO))
label_index = len(dataset.columns) -1


# prepare data
train_X, train_Y = data_to_sequences_and_labels(dataset.to_numpy(), params.LOOKBACK, 
                                                params.STEP, 
                                                0, train_max_idx, 
                                                params.DELAY,
                                                label_index) 
val_X, val_Y = data_to_sequences_and_labels(dataset.to_numpy(), params.LOOKBACK, 
                                                params.STEP, 
                                                train_max_idx+1, val_max_idx, 
                                                params.DELAY,
                                                label_index)
test_X, test_Y = data_to_sequences_and_labels(dataset.to_numpy(), params.LOOKBACK, 
                                                params.STEP, 
                                                val_max_idx+1, None, 
                                                params.DELAY,
                                                label_index)
X, Y = data_to_sequences_and_labels(dataset.to_numpy(), params.LOOKBACK,
                                   params.STEP,
                                   0, val_max_idx,
                                   params.DELAY
                                   ,label_index)

In [5]:
def lstm_model(input_shape, hidden_unit=32, n_layer=2, l2_weight=0.01):
    model = Sequential()
    is_return_sequences = True if n_layer>=2 else False
    
    # first layer
    model.add(LSTM(hidden_unit, 
                   input_shape=input_shape,
                   kernel_regularizer=regularizers.l2(l2_weight),
                   recurrent_regularizer=regularizers.l2(l2_weight),
                   return_sequences=is_return_sequences))
    
    # intermediate layers
    if n_layer >= 3:
        for _ in range(n_layer-2):
            model.add(LSTM(hidden_unit,
                           kernel_regularizer=regularizers.l2(l2_weight),
                           recurrent_regularizer=regularizers.l2(l2_weight),
                           return_sequences=is_return_sequences))

    
    # last layer
    if n_layer >= 2:
        model.add(LSTM(hidden_unit,
                       recurrent_regularizer=regularizers.l2(l2_weight),
                       kernel_regularizer=regularizers.l2(l2_weight)))
    
    model.add(Dense(1, activation='sigmoid'))
    model.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])
    return model

In [8]:
#Prepare for gridsearchCV
from sklearn.metrics import matthews_corrcoef, make_scorer
from sklearn.model_selection import TimeSeriesSplit, GridSearchCV 
cv = TimeSeriesSplit(n_splits=2)
matthews_score = make_scorer(matthews_corrcoef, greater_is_better=True)


In [31]:
lstm_params = dict(hidden_unit=[48, 64], 
                 n_layer=[3,4], 
                 epochs=[100, 250],
                 l2_weight=[0.001, 0.01])

# lstm_params = dict(hidden_unit=[64], 
#                  n_layer=[3], 
#                  epochs=[100],
#                  l2_weight=[0])

lstm_clf = GridSearchCV(estimator=KerasClassifier(build_fn=lstm_model, input_shape=INPUT_SHAPE, verbose=1),
                      param_grid=lstm_params,
                      cv =cv,
                      scoring=matthews_score)
grid_result=lstm_clf.fit(X,Y)

Epoch 1/100
1634/1634 [==============================] - 7s 4ms/step - loss: 0.8832 - acc: 0.6769
Epoch 2/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.7340 - acc: 0.6769
Epoch 3/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6563 - acc: 0.6830
Epoch 4/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6109 - acc: 0.7118
Epoch 5/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6056 - acc: 0.6928
Epoch 6/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5803 - acc: 0.7087
Epoch 7/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5574 - acc: 0.7289
Epoch 8/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5594 - acc: 0.7313
Epoch 9/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5661 - acc: 0.7160
Epoch 10/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5474 - acc: 0.7399
Epoch 11/100
1634/1

1634/1634 [==============================] - 2s 1ms/step - loss: 0.2393 - acc: 0.9284
Epoch 84/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2525 - acc: 0.9186
Epoch 85/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2300 - acc: 0.9315
Epoch 86/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2349 - acc: 0.9327
Epoch 87/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2127 - acc: 0.9394
Epoch 88/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2336 - acc: 0.9345
Epoch 89/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2410 - acc: 0.9253
Epoch 90/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1846 - acc: 0.9547
Epoch 91/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2454 - acc: 0.9204
Epoch 92/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1908 - acc: 0.9480
Epoch 93/100
1634/1634 

3267/3267 [==============================] - 3s 887us/step - loss: 0.2708 - acc: 0.9039
Epoch 65/100
3267/3267 [==============================] - 3s 887us/step - loss: 0.2638 - acc: 0.9063
Epoch 66/100
3267/3267 [==============================] - 3s 883us/step - loss: 0.2621 - acc: 0.9091
Epoch 67/100
3267/3267 [==============================] - 3s 882us/step - loss: 0.3105 - acc: 0.8871
Epoch 68/100
3267/3267 [==============================] - 3s 887us/step - loss: 0.2625 - acc: 0.9115
Epoch 69/100
3267/3267 [==============================] - 3s 879us/step - loss: 0.2592 - acc: 0.9140
Epoch 70/100
3267/3267 [==============================] - 3s 880us/step - loss: 0.2680 - acc: 0.9042
Epoch 71/100
3267/3267 [==============================] - 3s 881us/step - loss: 0.2529 - acc: 0.9128
Epoch 72/100
3267/3267 [==============================] - 3s 888us/step - loss: 0.2445 - acc: 0.9164
Epoch 73/100
3267/3267 [==============================] - 3s 891us/step - loss: 0.2539 - acc: 0.9094
Epo

1634/1634 [==============================] - 2s 1ms/step - loss: 0.4900 - acc: 0.7785
Epoch 46/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4928 - acc: 0.7693
Epoch 47/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4728 - acc: 0.7852
Epoch 48/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5081 - acc: 0.7576
Epoch 49/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4642 - acc: 0.7938
Epoch 50/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4766 - acc: 0.7778
Epoch 51/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4794 - acc: 0.7803
Epoch 52/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4614 - acc: 0.7950
Epoch 53/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4500 - acc: 0.8097
Epoch 54/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4753 - acc: 0.7895
Epoch 55/100
1634/1634 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.4608 - acc: 0.8032
Epoch 28/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4605 - acc: 0.8038
Epoch 29/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4795 - acc: 0.7903
Epoch 30/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4502 - acc: 0.8124
Epoch 31/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4650 - acc: 0.8059
Epoch 32/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4660 - acc: 0.8013
Epoch 33/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4463 - acc: 0.8194
Epoch 34/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4557 - acc: 0.8087
Epoch 35/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4449 - acc: 0.8185
Epoch 36/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4619 - acc: 0.8023
Epoch 37/100
3267/3267 

1634/1634 [==============================] - 1s 898us/step - loss: 0.6196 - acc: 0.6842
Epoch 10/100
1634/1634 [==============================] - 1s 900us/step - loss: 0.6051 - acc: 0.6848
Epoch 11/100
1634/1634 [==============================] - 1s 895us/step - loss: 0.5879 - acc: 0.6909
Epoch 12/100
1634/1634 [==============================] - 1s 901us/step - loss: 0.5744 - acc: 0.6977
Epoch 13/100
1634/1634 [==============================] - 1s 907us/step - loss: 0.5646 - acc: 0.7130
Epoch 14/100
1634/1634 [==============================] - 1s 893us/step - loss: 0.5559 - acc: 0.7148
Epoch 15/100
1634/1634 [==============================] - 1s 893us/step - loss: 0.5668 - acc: 0.7277
Epoch 16/100
1634/1634 [==============================] - 1s 895us/step - loss: 0.5585 - acc: 0.7124
Epoch 17/100
1634/1634 [==============================] - 1s 891us/step - loss: 0.5499 - acc: 0.7179
Epoch 18/100
1634/1634 [==============================] - 1s 896us/step - loss: 0.5469 - acc: 0.7191
Epo

1634/1634 [==============================] - 1s 902us/step - loss: 0.5140 - acc: 0.7424
Epoch 91/100
1634/1634 [==============================] - 1s 902us/step - loss: 0.4980 - acc: 0.7583
Epoch 92/100
1634/1634 [==============================] - 1s 892us/step - loss: 0.5074 - acc: 0.7534
Epoch 93/100
1634/1634 [==============================] - 1s 896us/step - loss: 0.4992 - acc: 0.7515
Epoch 94/100
1634/1634 [==============================] - 1s 898us/step - loss: 0.4962 - acc: 0.7589
Epoch 95/100
1634/1634 [==============================] - 1s 906us/step - loss: 0.5110 - acc: 0.7417
Epoch 96/100
1634/1634 [==============================] - 1s 893us/step - loss: 0.5072 - acc: 0.7472
Epoch 97/100
1634/1634 [==============================] - 1s 899us/step - loss: 0.5059 - acc: 0.7472
Epoch 98/100
1634/1634 [==============================] - 1s 896us/step - loss: 0.5235 - acc: 0.7307
Epoch 99/100
1634/1634 [==============================] - 1s 903us/step - loss: 0.5129 - acc: 0.7442
Epo

3267/3267 [==============================] - 3s 896us/step - loss: 0.4708 - acc: 0.7912
Epoch 71/100
3267/3267 [==============================] - 3s 889us/step - loss: 0.4798 - acc: 0.7830
Epoch 72/100
3267/3267 [==============================] - 3s 890us/step - loss: 0.4725 - acc: 0.7903
Epoch 73/100
3267/3267 [==============================] - 3s 893us/step - loss: 0.4693 - acc: 0.7879
Epoch 74/100
3267/3267 [==============================] - 3s 892us/step - loss: 0.4774 - acc: 0.7891
Epoch 75/100
3267/3267 [==============================] - 3s 887us/step - loss: 0.4727 - acc: 0.7876
Epoch 76/100
3267/3267 [==============================] - 3s 891us/step - loss: 0.4750 - acc: 0.7821
Epoch 77/100
3267/3267 [==============================] - 3s 891us/step - loss: 0.4719 - acc: 0.7882
Epoch 78/100
3267/3267 [==============================] - 3s 892us/step - loss: 0.4747 - acc: 0.7931
Epoch 79/100
3267/3267 [==============================] - 3s 890us/step - loss: 0.4748 - acc: 0.7894
Epo

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6309 - acc: 0.6775
Epoch 52/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 53/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 54/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6305 - acc: 0.6775
Epoch 55/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 56/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6289 - acc: 0.6775
Epoch 57/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 58/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 59/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 60/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6297 - acc: 0.6775
Epoch 61/100
1634/1634 

/home/kevin/anaconda2/envs/py36/lib/python3.6/site-packages/sklearn/metrics/classification.py:872: RuntimeWarning: invalid value encountered in double_scalars
  mcc = cov_ytyp / np.sqrt(cov_ytyt * cov_ypyp)


Epoch 1/100
3267/3267 [==============================] - 11s 3ms/step - loss: 2.6013 - acc: 0.7205
Epoch 2/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.8077 - acc: 0.7224
Epoch 3/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.6177 - acc: 0.7224
Epoch 4/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5974 - acc: 0.7224
Epoch 5/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5944 - acc: 0.7224
Epoch 6/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5922 - acc: 0.7224
Epoch 7/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 8/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 9/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5930 - acc: 0.7224
Epoch 10/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 11/100
3267/

3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 84/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 85/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 86/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5909 - acc: 0.7224
Epoch 87/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 88/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 89/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5919 - acc: 0.7224
Epoch 90/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 91/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 92/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5917 - acc: 0.7224
Epoch 93/100
3267/3267 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.3141 - acc: 0.8953
Epoch 66/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2835 - acc: 0.9131
Epoch 67/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3221 - acc: 0.8972
Epoch 68/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2983 - acc: 0.9039
Epoch 69/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3032 - acc: 0.9002
Epoch 70/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3150 - acc: 0.8935
Epoch 71/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2951 - acc: 0.9088
Epoch 72/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2579 - acc: 0.9241
Epoch 73/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2596 - acc: 0.9119
Epoch 74/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2549 - acc: 0.9235
Epoch 75/100
1634/1634 

3267/3267 [==============================] - 3s 1ms/step - loss: 0.2800 - acc: 0.9008
Epoch 48/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2683 - acc: 0.9073
Epoch 49/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2856 - acc: 0.8981
Epoch 50/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2912 - acc: 0.8965
Epoch 51/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2661 - acc: 0.9106
Epoch 52/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2768 - acc: 0.9063
Epoch 53/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2803 - acc: 0.9024
Epoch 54/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2926 - acc: 0.8987
Epoch 55/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.3149 - acc: 0.8867
Epoch 56/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2654 - acc: 0.9112
Epoch 57/100
3267/3267 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.4968 - acc: 0.7687
Epoch 30/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4992 - acc: 0.7564
Epoch 31/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4991 - acc: 0.7625
Epoch 32/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4842 - acc: 0.7742
Epoch 33/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4844 - acc: 0.7803
Epoch 34/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5066 - acc: 0.7583
Epoch 35/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4944 - acc: 0.7748
Epoch 36/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4923 - acc: 0.7650
Epoch 37/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5189 - acc: 0.7491
Epoch 38/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5160 - acc: 0.7497
Epoch 39/100
1634/1634 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.5065 - acc: 0.7713
Epoch 12/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4929 - acc: 0.7842
Epoch 13/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4974 - acc: 0.7796
Epoch 14/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4868 - acc: 0.7839
Epoch 15/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4891 - acc: 0.7799
Epoch 16/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4922 - acc: 0.7906
Epoch 17/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4860 - acc: 0.7870
Epoch 18/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4893 - acc: 0.7860
Epoch 19/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4756 - acc: 0.7989
Epoch 20/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4764 - acc: 0.7916
Epoch 21/100
3267/3267 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.2761 - acc: 0.9094
Epoch 94/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2612 - acc: 0.9143
Epoch 95/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2612 - acc: 0.9183
Epoch 96/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2529 - acc: 0.9226
Epoch 97/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2919 - acc: 0.9069
Epoch 98/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2579 - acc: 0.9189
Epoch 99/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2542 - acc: 0.9177
Epoch 100/100
1633/1633 [==============================] - 4s 2ms/step
Epoch 1/100
1634/1634 [==============================] - 9s 6ms/step - loss: 3.3072 - acc: 0.6714
Epoch 2/100
1634/1634 [==============================] - 2s 1ms/step - loss: 1.3513 - acc: 0.6775
Epoch 3/100
1634/1634 [==============================

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 76/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6297 - acc: 0.6775
Epoch 77/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6290 - acc: 0.6775
Epoch 78/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6302 - acc: 0.6775
Epoch 79/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 80/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 81/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6301 - acc: 0.6775
Epoch 82/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 83/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 84/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 85/100
1634/1634 

3267/3267 [==============================] - 3s 1ms/step - loss: 0.4729 - acc: 0.7845
Epoch 58/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4937 - acc: 0.7769
Epoch 59/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4921 - acc: 0.7796
Epoch 60/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4779 - acc: 0.7839
Epoch 61/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4856 - acc: 0.7821
Epoch 62/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4816 - acc: 0.7854
Epoch 63/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4862 - acc: 0.7778
Epoch 64/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4827 - acc: 0.7781
Epoch 65/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4818 - acc: 0.7851
Epoch 66/100
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4810 - acc: 0.7833
Epoch 67/100
3267/3267 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6303 - acc: 0.6775
Epoch 40/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 41/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 42/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 43/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 44/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 45/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 46/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 47/100
1634/1634 [==============================] - 3s 2ms/step - loss: 0.6293 - acc: 0.6775
Epoch 48/100
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6303 - acc: 0.6775
Epoch 49/100
1634/1634 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.5917 - acc: 0.7224
Epoch 22/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 23/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 24/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5917 - acc: 0.7224
Epoch 25/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5923 - acc: 0.7224
Epoch 26/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5921 - acc: 0.7224
Epoch 27/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 28/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5921 - acc: 0.7224
Epoch 29/100
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5916 - acc: 0.7224
Epoch 30/100
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5922 - acc: 0.7224
Epoch 31/100
3267/3267 

1634/1634 [==============================] - 2s 931us/step - loss: 0.6901 - acc: 0.6781
Epoch 4/250
1634/1634 [==============================] - 2s 924us/step - loss: 0.6295 - acc: 0.6897
Epoch 5/250
1634/1634 [==============================] - 2s 935us/step - loss: 0.6020 - acc: 0.7062
Epoch 6/250
1634/1634 [==============================] - 2s 920us/step - loss: 0.5691 - acc: 0.7277
Epoch 7/250
1634/1634 [==============================] - 2s 926us/step - loss: 0.5757 - acc: 0.7087
Epoch 8/250
1634/1634 [==============================] - 2s 928us/step - loss: 0.5566 - acc: 0.7319
Epoch 9/250
1634/1634 [==============================] - 2s 927us/step - loss: 0.5643 - acc: 0.7258
Epoch 10/250
1634/1634 [==============================] - 2s 923us/step - loss: 0.5444 - acc: 0.7393
Epoch 11/250
1634/1634 [==============================] - 2s 926us/step - loss: 0.5286 - acc: 0.7687
Epoch 12/250
1634/1634 [==============================] - 2s 920us/step - loss: 0.5454 - acc: 0.7411
Epoch 13/

1634/1634 [==============================] - 2s 918us/step - loss: 0.2429 - acc: 0.9229
Epoch 85/250
1634/1634 [==============================] - 2s 926us/step - loss: 0.2381 - acc: 0.9253
Epoch 86/250
1634/1634 [==============================] - 1s 917us/step - loss: 0.2001 - acc: 0.9394
Epoch 87/250
1634/1634 [==============================] - 2s 924us/step - loss: 0.2362 - acc: 0.9278
Epoch 88/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.2068 - acc: 0.9443
Epoch 89/250
1634/1634 [==============================] - 2s 923us/step - loss: 0.2003 - acc: 0.9443
Epoch 90/250
1634/1634 [==============================] - 2s 924us/step - loss: 0.2242 - acc: 0.9266
Epoch 91/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.1949 - acc: 0.9510
Epoch 92/250
1634/1634 [==============================] - 1s 918us/step - loss: 0.2403 - acc: 0.9333
Epoch 93/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.1925 - acc: 0.9419
Epo

1634/1634 [==============================] - 2s 919us/step - loss: 0.1437 - acc: 0.9694
Epoch 165/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.1226 - acc: 0.9706
Epoch 166/250
1634/1634 [==============================] - 1s 916us/step - loss: 0.1049 - acc: 0.9829
Epoch 167/250
1634/1634 [==============================] - 2s 932us/step - loss: 0.1413 - acc: 0.9670
Epoch 168/250
1634/1634 [==============================] - 1s 918us/step - loss: 0.1213 - acc: 0.9792
Epoch 169/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.1374 - acc: 0.9682
Epoch 170/250
1634/1634 [==============================] - 2s 920us/step - loss: 0.1353 - acc: 0.9682
Epoch 171/250
1634/1634 [==============================] - 2s 922us/step - loss: 0.1326 - acc: 0.9706
Epoch 172/250
1634/1634 [==============================] - 2s 920us/step - loss: 0.1103 - acc: 0.9816
Epoch 173/250
1634/1634 [==============================] - 2s 926us/step - loss: 0.1735 - acc: 0

1634/1634 [==============================] - 2s 927us/step - loss: 0.2369 - acc: 0.9357
Epoch 245/250
1634/1634 [==============================] - 2s 919us/step - loss: 0.1191 - acc: 0.9786
Epoch 246/250
1634/1634 [==============================] - 2s 935us/step - loss: 0.1771 - acc: 0.9559
Epoch 247/250
1634/1634 [==============================] - 2s 919us/step - loss: 0.1084 - acc: 0.9774
Epoch 248/250
1634/1634 [==============================] - 2s 924us/step - loss: 0.0964 - acc: 0.9853
Epoch 249/250
1634/1634 [==============================] - 2s 922us/step - loss: 0.1236 - acc: 0.9731
Epoch 250/250
1633/1633 [==============================] - 4s 3ms/step
Epoch 1/250
3267/3267 [==============================] - 12s 4ms/step - loss: 0.7932 - acc: 0.7251
Epoch 2/250
3267/3267 [==============================] - 3s 915us/step - loss: 0.6093 - acc: 0.7481
Epoch 3/250
3267/3267 [==============================] - 3s 915us/step - loss: 0.5836 - acc: 0.7371
Epoch 4/250
3267/3267 [=========

3267/3267 [==============================] - 3s 911us/step - loss: 0.2339 - acc: 0.9247
Epoch 75/250
3267/3267 [==============================] - 3s 915us/step - loss: 0.2170 - acc: 0.9324
Epoch 76/250
3267/3267 [==============================] - 3s 912us/step - loss: 0.2334 - acc: 0.9262
Epoch 77/250
3267/3267 [==============================] - 3s 911us/step - loss: 0.2278 - acc: 0.9287
Epoch 78/250
3267/3267 [==============================] - 3s 925us/step - loss: 0.2258 - acc: 0.9317
Epoch 79/250
3267/3267 [==============================] - 3s 920us/step - loss: 0.2254 - acc: 0.9330
Epoch 80/250
3267/3267 [==============================] - 3s 968us/step - loss: 0.2270 - acc: 0.9308
Epoch 81/250
3267/3267 [==============================] - 3s 911us/step - loss: 0.2047 - acc: 0.9394
Epoch 82/250
3267/3267 [==============================] - 3s 915us/step - loss: 0.2120 - acc: 0.9385
Epoch 83/250
3267/3267 [==============================] - 3s 925us/step - loss: 0.2183 - acc: 0.9388
Epo

3267/3267 [==============================] - 3s 916us/step - loss: 0.1424 - acc: 0.9691
Epoch 155/250
3267/3267 [==============================] - 3s 911us/step - loss: 0.1276 - acc: 0.9721
Epoch 156/250
3267/3267 [==============================] - 3s 913us/step - loss: 0.1234 - acc: 0.9740
Epoch 157/250
3267/3267 [==============================] - 3s 910us/step - loss: 0.1482 - acc: 0.9642
Epoch 158/250
3267/3267 [==============================] - 3s 925us/step - loss: 0.1564 - acc: 0.9611
Epoch 159/250
3267/3267 [==============================] - 3s 916us/step - loss: 0.1321 - acc: 0.9685
Epoch 160/250
3267/3267 [==============================] - 3s 911us/step - loss: 0.1416 - acc: 0.9685
Epoch 161/250
3267/3267 [==============================] - 3s 914us/step - loss: 0.1686 - acc: 0.9590
Epoch 162/250
3267/3267 [==============================] - 3s 913us/step - loss: 0.1191 - acc: 0.9749
Epoch 163/250
3267/3267 [==============================] - 3s 927us/step - loss: 0.1315 - acc: 0

3267/3267 [==============================] - 3s 913us/step - loss: 0.1179 - acc: 0.9749
Epoch 235/250
3267/3267 [==============================] - 3s 913us/step - loss: 0.1419 - acc: 0.9672
Epoch 236/250
3267/3267 [==============================] - 3s 911us/step - loss: 0.1037 - acc: 0.9804
Epoch 237/250
3267/3267 [==============================] - 3s 912us/step - loss: 0.1003 - acc: 0.9804
Epoch 238/250
3267/3267 [==============================] - 3s 926us/step - loss: 0.1063 - acc: 0.9780
Epoch 239/250
3267/3267 [==============================] - 3s 913us/step - loss: 0.1341 - acc: 0.9676
Epoch 240/250
3267/3267 [==============================] - 3s 912us/step - loss: 0.1291 - acc: 0.9731
Epoch 241/250
3267/3267 [==============================] - 3s 917us/step - loss: 0.1122 - acc: 0.9786
Epoch 242/250
3267/3267 [==============================] - 3s 914us/step - loss: 0.1112 - acc: 0.9767
Epoch 243/250
3267/3267 [==============================] - 3s 914us/step - loss: 0.0999 - acc: 0

1634/1634 [==============================] - 2s 1ms/step - loss: 0.4788 - acc: 0.7729
Epoch 66/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4659 - acc: 0.7852
Epoch 67/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4639 - acc: 0.7864
Epoch 68/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4862 - acc: 0.7711
Epoch 69/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4505 - acc: 0.7840
Epoch 70/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4409 - acc: 0.7944
Epoch 71/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4472 - acc: 0.7956
Epoch 72/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4396 - acc: 0.8017
Epoch 73/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4495 - acc: 0.8011
Epoch 74/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4609 - acc: 0.7925
Epoch 75/250
1634/1634 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.3527 - acc: 0.8605
Epoch 148/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3588 - acc: 0.8660
Epoch 149/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3632 - acc: 0.8672
Epoch 150/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4429 - acc: 0.8188
Epoch 151/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4659 - acc: 0.7846
Epoch 152/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3872 - acc: 0.8476
Epoch 153/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3900 - acc: 0.8470
Epoch 154/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3791 - acc: 0.8550
Epoch 155/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3942 - acc: 0.8348
Epoch 156/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3609 - acc: 0.8684
Epoch 157/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.2824 - acc: 0.8941
Epoch 230/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2992 - acc: 0.8947
Epoch 231/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3232 - acc: 0.8733
Epoch 232/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2874 - acc: 0.8947
Epoch 233/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2818 - acc: 0.8960
Epoch 234/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2814 - acc: 0.8966
Epoch 235/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2907 - acc: 0.8917
Epoch 236/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3222 - acc: 0.8770
Epoch 237/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2868 - acc: 0.8978
Epoch 238/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2954 - acc: 0.8941
Epoch 239/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.4044 - acc: 0.8463
Epoch 62/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4058 - acc: 0.8408
Epoch 63/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3984 - acc: 0.8421
Epoch 64/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4008 - acc: 0.8460
Epoch 65/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3958 - acc: 0.8485
Epoch 66/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3871 - acc: 0.8512
Epoch 67/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3790 - acc: 0.8512
Epoch 68/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3881 - acc: 0.8448
Epoch 69/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3824 - acc: 0.8512
Epoch 70/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.3858 - acc: 0.8479
Epoch 71/250
3267/3267 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.2078 - acc: 0.9403
Epoch 144/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2112 - acc: 0.9369
Epoch 145/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2297 - acc: 0.9290
Epoch 146/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2045 - acc: 0.9418
Epoch 147/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2305 - acc: 0.9284
Epoch 148/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2018 - acc: 0.9421
Epoch 149/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2031 - acc: 0.9400
Epoch 150/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1894 - acc: 0.9434
Epoch 151/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2244 - acc: 0.9354
Epoch 152/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2613 - acc: 0.9180
Epoch 153/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.1483 - acc: 0.9614
Epoch 226/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1703 - acc: 0.9519
Epoch 227/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1804 - acc: 0.9526
Epoch 228/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1584 - acc: 0.9575
Epoch 229/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1519 - acc: 0.9620
Epoch 230/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1616 - acc: 0.9602
Epoch 231/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1664 - acc: 0.9541
Epoch 232/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1575 - acc: 0.9620
Epoch 233/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1399 - acc: 0.9660
Epoch 234/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1813 - acc: 0.9486
Epoch 235/250


1634/1634 [==============================] - 2s 930us/step - loss: 0.5111 - acc: 0.7497
Epoch 57/250
1634/1634 [==============================] - 2s 935us/step - loss: 0.5129 - acc: 0.7436
Epoch 58/250
1634/1634 [==============================] - 2s 924us/step - loss: 0.5158 - acc: 0.7252
Epoch 59/250
1634/1634 [==============================] - 2s 932us/step - loss: 0.5004 - acc: 0.7521
Epoch 60/250
1634/1634 [==============================] - 2s 925us/step - loss: 0.5037 - acc: 0.7515
Epoch 61/250
1634/1634 [==============================] - 2s 944us/step - loss: 0.5052 - acc: 0.7509
Epoch 62/250
1634/1634 [==============================] - 2s 926us/step - loss: 0.5144 - acc: 0.7344
Epoch 63/250
1634/1634 [==============================] - 2s 939us/step - loss: 0.5134 - acc: 0.7393
Epoch 64/250
1634/1634 [==============================] - 2s 929us/step - loss: 0.5298 - acc: 0.7111
Epoch 65/250
1634/1634 [==============================] - 2s 935us/step - loss: 0.5003 - acc: 0.7521
Epo

1634/1634 [==============================] - 2s 927us/step - loss: 0.4837 - acc: 0.7576
Epoch 137/250
1634/1634 [==============================] - 2s 932us/step - loss: 0.4955 - acc: 0.7509
Epoch 138/250
1634/1634 [==============================] - 2s 931us/step - loss: 0.4835 - acc: 0.7638
Epoch 139/250
1634/1634 [==============================] - 2s 936us/step - loss: 0.4779 - acc: 0.7638
Epoch 140/250
1634/1634 [==============================] - 2s 930us/step - loss: 0.4907 - acc: 0.7595
Epoch 141/250
1634/1634 [==============================] - 2s 933us/step - loss: 0.4885 - acc: 0.7564
Epoch 142/250
1634/1634 [==============================] - 2s 933us/step - loss: 0.4886 - acc: 0.7632
Epoch 143/250
1634/1634 [==============================] - 2s 933us/step - loss: 0.4923 - acc: 0.7540
Epoch 144/250
1634/1634 [==============================] - 2s 935us/step - loss: 0.4826 - acc: 0.7681
Epoch 145/250
1634/1634 [==============================] - 2s 934us/step - loss: 0.4901 - acc: 0

1634/1634 [==============================] - 2s 934us/step - loss: 0.4978 - acc: 0.7613
Epoch 217/250
1634/1634 [==============================] - 2s 929us/step - loss: 0.5795 - acc: 0.6695
Epoch 218/250
1634/1634 [==============================] - 2s 929us/step - loss: 0.5125 - acc: 0.7105
Epoch 219/250
1634/1634 [==============================] - 2s 930us/step - loss: 0.5063 - acc: 0.7552
Epoch 220/250
1634/1634 [==============================] - 2s 930us/step - loss: 0.4834 - acc: 0.7662
Epoch 221/250
1634/1634 [==============================] - 2s 933us/step - loss: 0.4794 - acc: 0.7693
Epoch 222/250
1634/1634 [==============================] - 2s 950us/step - loss: 0.4698 - acc: 0.7736
Epoch 223/250
1634/1634 [==============================] - 2s 931us/step - loss: 0.4761 - acc: 0.7687
Epoch 224/250
1634/1634 [==============================] - 2s 932us/step - loss: 0.4788 - acc: 0.7717
Epoch 225/250
1634/1634 [==============================] - 2s 927us/step - loss: 0.4808 - acc: 0

3267/3267 [==============================] - 3s 929us/step - loss: 0.4877 - acc: 0.7909
Epoch 47/250
3267/3267 [==============================] - 3s 923us/step - loss: 0.4881 - acc: 0.7833
Epoch 48/250
3267/3267 [==============================] - 3s 925us/step - loss: 0.4898 - acc: 0.7802
Epoch 49/250
3267/3267 [==============================] - 3s 922us/step - loss: 0.4826 - acc: 0.7854
Epoch 50/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4838 - acc: 0.7885
Epoch 51/250
3267/3267 [==============================] - 3s 920us/step - loss: 0.4786 - acc: 0.7842
Epoch 52/250
3267/3267 [==============================] - 3s 926us/step - loss: 0.4787 - acc: 0.7867
Epoch 53/250
3267/3267 [==============================] - 3s 924us/step - loss: 0.4868 - acc: 0.7833
Epoch 54/250
3267/3267 [==============================] - 3s 922us/step - loss: 0.4857 - acc: 0.7799
Epoch 55/250
3267/3267 [==============================] - 3s 924us/step - loss: 0.4818 - acc: 0.7857
Epo

3267/3267 [==============================] - 3s 924us/step - loss: 0.4531 - acc: 0.8118
Epoch 128/250
3267/3267 [==============================] - 3s 920us/step - loss: 0.4526 - acc: 0.8078
Epoch 129/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4646 - acc: 0.8017
Epoch 130/250
3267/3267 [==============================] - 3s 920us/step - loss: 0.4630 - acc: 0.8084
Epoch 131/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4513 - acc: 0.8090
Epoch 132/250
3267/3267 [==============================] - 3s 927us/step - loss: 0.4465 - acc: 0.8121
Epoch 133/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4597 - acc: 0.8038
Epoch 134/250
3267/3267 [==============================] - 3s 923us/step - loss: 0.4523 - acc: 0.8114
Epoch 135/250
3267/3267 [==============================] - 3s 923us/step - loss: 0.4530 - acc: 0.8087
Epoch 136/250
3267/3267 [==============================] - 3s 922us/step - loss: 0.4512 - acc: 0

3267/3267 [==============================] - 3s 921us/step - loss: 0.4433 - acc: 0.8102
Epoch 208/250
3267/3267 [==============================] - 3s 923us/step - loss: 0.4410 - acc: 0.8176
Epoch 209/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4490 - acc: 0.8081
Epoch 210/250
3267/3267 [==============================] - 3s 924us/step - loss: 0.4486 - acc: 0.8069
Epoch 211/250
3267/3267 [==============================] - 3s 921us/step - loss: 0.4417 - acc: 0.8124
Epoch 212/250
3267/3267 [==============================] - 3s 929us/step - loss: 0.4468 - acc: 0.8066
Epoch 213/250
3267/3267 [==============================] - 3s 922us/step - loss: 0.4483 - acc: 0.8111
Epoch 214/250
3267/3267 [==============================] - 3s 925us/step - loss: 0.4431 - acc: 0.8121
Epoch 215/250
3267/3267 [==============================] - 3s 923us/step - loss: 0.4496 - acc: 0.8050
Epoch 216/250
3267/3267 [==============================] - 3s 926us/step - loss: 0.4410 - acc: 0

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 39/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 40/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 41/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 42/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 43/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 44/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 45/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 46/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6304 - acc: 0.6775
Epoch 47/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6312 - acc: 0.6775
Epoch 48/250
1634/1634 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6300 - acc: 0.6775
Epoch 121/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 122/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 123/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 124/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 125/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 126/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 127/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 128/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 129/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 130/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.6297 - acc: 0.6775
Epoch 203/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6290 - acc: 0.6775
Epoch 204/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6289 - acc: 0.6775
Epoch 205/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6300 - acc: 0.6775
Epoch 206/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6310 - acc: 0.6775
Epoch 207/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6300 - acc: 0.6775
Epoch 208/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 209/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 210/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6289 - acc: 0.6775
Epoch 211/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 212/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 34/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 35/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 36/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5908 - acc: 0.7224
Epoch 37/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 38/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 39/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 40/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 41/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5917 - acc: 0.7224
Epoch 42/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 43/250
3267/3267 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.5919 - acc: 0.7224
Epoch 116/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 117/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 118/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5909 - acc: 0.7224
Epoch 119/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 120/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 121/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 122/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 123/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 124/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5909 - acc: 0.7224
Epoch 125/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 198/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 199/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5920 - acc: 0.7224
Epoch 200/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 201/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 202/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 203/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 204/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5917 - acc: 0.7224
Epoch 205/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 206/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5909 - acc: 0.7224
Epoch 207/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.6372 - acc: 0.7313
Epoch 29/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5220 - acc: 0.7846
Epoch 30/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4690 - acc: 0.8133
Epoch 31/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4717 - acc: 0.8213
Epoch 32/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4429 - acc: 0.8299
Epoch 33/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4148 - acc: 0.8415
Epoch 34/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4021 - acc: 0.8464
Epoch 35/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3840 - acc: 0.8605
Epoch 36/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4174 - acc: 0.8439
Epoch 37/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3928 - acc: 0.8537
Epoch 38/250
1634/1634 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.1637 - acc: 0.9621
Epoch 111/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1428 - acc: 0.9700
Epoch 112/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1940 - acc: 0.9541
Epoch 113/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1543 - acc: 0.9590
Epoch 114/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3299 - acc: 0.9027
Epoch 115/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2389 - acc: 0.9364
Epoch 116/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1524 - acc: 0.9614
Epoch 117/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1619 - acc: 0.9627
Epoch 118/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1450 - acc: 0.9670
Epoch 119/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1563 - acc: 0.9651
Epoch 120/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.1425 - acc: 0.9670
Epoch 193/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1172 - acc: 0.9761
Epoch 194/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1092 - acc: 0.9816
Epoch 195/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.0945 - acc: 0.9865
Epoch 196/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1009 - acc: 0.9841
Epoch 197/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1115 - acc: 0.9755
Epoch 198/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2416 - acc: 0.9315
Epoch 199/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1119 - acc: 0.9810
Epoch 200/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1188 - acc: 0.9731
Epoch 201/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.1012 - acc: 0.9835
Epoch 202/250


3267/3267 [==============================] - 3s 1ms/step - loss: 0.4398 - acc: 0.8121
Epoch 24/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4411 - acc: 0.8154
Epoch 25/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4099 - acc: 0.8307
Epoch 26/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4238 - acc: 0.8228
Epoch 27/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.4172 - acc: 0.8261
Epoch 28/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.3945 - acc: 0.8350
Epoch 29/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.3896 - acc: 0.8365
Epoch 30/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4170 - acc: 0.8206
Epoch 31/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.3987 - acc: 0.8369
Epoch 32/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.3979 - acc: 0.8424
Epoch 33/250
3267/3267 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.1983 - acc: 0.9421
Epoch 106/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1827 - acc: 0.9489
Epoch 107/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1839 - acc: 0.9474
Epoch 108/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1758 - acc: 0.9519
Epoch 109/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2740 - acc: 0.9134
Epoch 110/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1950 - acc: 0.9421
Epoch 111/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1779 - acc: 0.9510
Epoch 112/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1732 - acc: 0.9501
Epoch 113/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.1682 - acc: 0.9535
Epoch 114/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.2004 - acc: 0.9418
Epoch 115/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.1263 - acc: 0.9685
Epoch 188/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1391 - acc: 0.9712
Epoch 189/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1392 - acc: 0.9679
Epoch 190/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1608 - acc: 0.9608
Epoch 191/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1183 - acc: 0.9755
Epoch 192/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1310 - acc: 0.9688
Epoch 193/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1316 - acc: 0.9715
Epoch 194/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.2595 - acc: 0.9210
Epoch 195/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1423 - acc: 0.9712
Epoch 196/250
3267/3267 [==============================] - 3s 1ms/step - loss: 0.1275 - acc: 0.9749
Epoch 197/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.5306 - acc: 0.7381
Epoch 19/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5344 - acc: 0.7124
Epoch 20/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6621 - acc: 0.6420
Epoch 21/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5892 - acc: 0.7020
Epoch 22/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5265 - acc: 0.7405
Epoch 23/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5401 - acc: 0.7013
Epoch 24/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5237 - acc: 0.7393
Epoch 25/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5189 - acc: 0.7344
Epoch 26/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5245 - acc: 0.7405
Epoch 27/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5122 - acc: 0.7417
Epoch 28/250
1634/1634 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.4351 - acc: 0.8054
Epoch 101/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4325 - acc: 0.8109
Epoch 102/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4379 - acc: 0.8060
Epoch 103/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4349 - acc: 0.8170
Epoch 104/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4275 - acc: 0.8201
Epoch 105/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4294 - acc: 0.8188
Epoch 106/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4254 - acc: 0.8127
Epoch 107/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4273 - acc: 0.8115
Epoch 108/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4402 - acc: 0.8176
Epoch 109/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4197 - acc: 0.8207
Epoch 110/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.3242 - acc: 0.8843
Epoch 183/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3381 - acc: 0.8703
Epoch 184/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3095 - acc: 0.8862
Epoch 185/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3354 - acc: 0.8776
Epoch 186/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3046 - acc: 0.8966
Epoch 187/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3257 - acc: 0.8800
Epoch 188/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3089 - acc: 0.8892
Epoch 189/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3296 - acc: 0.8831
Epoch 190/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.2813 - acc: 0.9076
Epoch 191/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.3245 - acc: 0.8825
Epoch 192/250


3267/3267 [==============================] - 5s 1ms/step - loss: 0.5003 - acc: 0.7713
Epoch 14/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5078 - acc: 0.7671
Epoch 15/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.4999 - acc: 0.7674
Epoch 16/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5043 - acc: 0.7701
Epoch 17/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.4928 - acc: 0.7726
Epoch 18/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.4917 - acc: 0.7815
Epoch 19/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.4951 - acc: 0.7781
Epoch 20/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.4823 - acc: 0.7857
Epoch 21/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5001 - acc: 0.7802
Epoch 22/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5021 - acc: 0.7738
Epoch 23/250
3267/3267 

3267/3267 [==============================] - 5s 1ms/step - loss: 0.2638 - acc: 0.9060
Epoch 96/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2463 - acc: 0.9134
Epoch 97/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2481 - acc: 0.9131
Epoch 98/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2489 - acc: 0.9125
Epoch 99/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2557 - acc: 0.9079
Epoch 100/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2410 - acc: 0.9073
Epoch 101/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2498 - acc: 0.9140
Epoch 102/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2392 - acc: 0.9195
Epoch 103/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2598 - acc: 0.9042
Epoch 104/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.2447 - acc: 0.9134
Epoch 105/250
3267

3267/3267 [==============================] - 5s 1ms/step - loss: 0.1575 - acc: 0.9611
Epoch 178/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1477 - acc: 0.9636
Epoch 179/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1646 - acc: 0.9565
Epoch 180/250
3267/3267 [==============================] - 5s 2ms/step - loss: 0.1468 - acc: 0.9657
Epoch 181/250
3267/3267 [==============================] - 6s 2ms/step - loss: 0.1629 - acc: 0.9544
Epoch 182/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1492 - acc: 0.9666
Epoch 183/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1389 - acc: 0.9682
Epoch 184/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1661 - acc: 0.9593
Epoch 185/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1731 - acc: 0.9565
Epoch 186/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.1568 - acc: 0.9593
Epoch 187/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.5998 - acc: 0.6848
Epoch 9/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5926 - acc: 0.6940
Epoch 10/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6392 - acc: 0.6781
Epoch 11/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5989 - acc: 0.6818
Epoch 12/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5740 - acc: 0.7148
Epoch 13/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5836 - acc: 0.7026
Epoch 14/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5953 - acc: 0.6824
Epoch 15/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5795 - acc: 0.7142
Epoch 16/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5556 - acc: 0.7142
Epoch 17/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5570 - acc: 0.7093
Epoch 18/250
1634/1634 [

1634/1634 [==============================] - 2s 1ms/step - loss: 0.5098 - acc: 0.7375
Epoch 91/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5162 - acc: 0.7515
Epoch 92/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5155 - acc: 0.7307
Epoch 93/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5021 - acc: 0.7472
Epoch 94/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5102 - acc: 0.7405
Epoch 95/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5103 - acc: 0.7405
Epoch 96/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5052 - acc: 0.7381
Epoch 97/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5015 - acc: 0.7417
Epoch 98/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4970 - acc: 0.7528
Epoch 99/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4984 - acc: 0.7546
Epoch 100/250
1634/1634

1634/1634 [==============================] - 2s 1ms/step - loss: 0.4508 - acc: 0.8133
Epoch 173/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4379 - acc: 0.8201
Epoch 174/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4388 - acc: 0.8176
Epoch 175/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4656 - acc: 0.8140
Epoch 176/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4561 - acc: 0.8103
Epoch 177/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.5671 - acc: 0.7295
Epoch 178/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4440 - acc: 0.8225
Epoch 179/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4544 - acc: 0.8140
Epoch 180/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4666 - acc: 0.8109
Epoch 181/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.4424 - acc: 0.8262
Epoch 182/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.5800 - acc: 0.7469
Epoch 4/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5537 - acc: 0.7530
Epoch 5/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5483 - acc: 0.7539
Epoch 6/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5429 - acc: 0.7524
Epoch 7/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5357 - acc: 0.7567
Epoch 8/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5386 - acc: 0.7570
Epoch 9/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5285 - acc: 0.7625
Epoch 10/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5338 - acc: 0.7570
Epoch 11/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5206 - acc: 0.7634
Epoch 12/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.5147 - acc: 0.7692
Epoch 13/250
3267/3267 [=====

3267/3267 [==============================] - 4s 1ms/step - loss: 0.4745 - acc: 0.7873
Epoch 86/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4729 - acc: 0.7906
Epoch 87/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4782 - acc: 0.7836
Epoch 88/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4716 - acc: 0.7863
Epoch 89/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4679 - acc: 0.7916
Epoch 90/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4736 - acc: 0.7900
Epoch 91/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4738 - acc: 0.7876
Epoch 92/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4737 - acc: 0.7848
Epoch 93/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4678 - acc: 0.7854
Epoch 94/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4664 - acc: 0.7928
Epoch 95/250
3267/3267 

3267/3267 [==============================] - 4s 1ms/step - loss: 0.4609 - acc: 0.7961
Epoch 168/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4682 - acc: 0.7882
Epoch 169/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4557 - acc: 0.7931
Epoch 170/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4740 - acc: 0.7775
Epoch 171/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4617 - acc: 0.7968
Epoch 172/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4616 - acc: 0.7876
Epoch 173/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4618 - acc: 0.7906
Epoch 174/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4568 - acc: 0.7928
Epoch 175/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4633 - acc: 0.7903
Epoch 176/250
3267/3267 [==============================] - 4s 1ms/step - loss: 0.4579 - acc: 0.7867
Epoch 177/250


3267/3267 [==============================] - 4s 1ms/step - loss: 0.4546 - acc: 0.8010
Epoch 250/250
1633/1633 [==============================] - 6s 4ms/step
Epoch 1/250
1634/1634 [==============================] - 15s 9ms/step - loss: 4.2741 - acc: 0.6750
Epoch 2/250
1634/1634 [==============================] - 2s 1ms/step - loss: 1.6034 - acc: 0.6775
Epoch 3/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.8755 - acc: 0.6775
Epoch 4/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6929 - acc: 0.6775
Epoch 5/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6468 - acc: 0.6775
Epoch 6/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6364 - acc: 0.6775
Epoch 7/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6318 - acc: 0.6775
Epoch 8/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6305 - acc: 0.6775
Epoch 9/250
1634/1634 [==============================] - 2

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 82/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6295 - acc: 0.6775
Epoch 83/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 84/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 85/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6301 - acc: 0.6775
Epoch 86/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 87/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6297 - acc: 0.6775
Epoch 88/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 89/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 90/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 91/250
1634/1634 

1634/1634 [==============================] - 2s 1ms/step - loss: 0.6292 - acc: 0.6775
Epoch 164/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 165/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6293 - acc: 0.6775
Epoch 166/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 167/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6298 - acc: 0.6775
Epoch 168/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6290 - acc: 0.6775
Epoch 169/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 170/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 171/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 172/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6296 - acc: 0.6775
Epoch 173/250


1634/1634 [==============================] - 2s 1ms/step - loss: 0.6291 - acc: 0.6775
Epoch 246/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6297 - acc: 0.6775
Epoch 247/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6294 - acc: 0.6775
Epoch 248/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6299 - acc: 0.6775
Epoch 249/250
1634/1634 [==============================] - 2s 1ms/step - loss: 0.6289 - acc: 0.6775
Epoch 250/250
1633/1633 [==============================] - 6s 4ms/step
Epoch 1/250
3267/3267 [==============================] - 17s 5ms/step - loss: 2.9449 - acc: 0.7224
Epoch 2/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.7636 - acc: 0.7224
Epoch 3/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.6079 - acc: 0.7224
Epoch 4/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5961 - acc: 0.7224
Epoch 5/250
3267/3267 [===========================

3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 78/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 79/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5915 - acc: 0.7224
Epoch 80/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5919 - acc: 0.7224
Epoch 81/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 82/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 83/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 84/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 85/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 86/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 87/250
3267/3267 

3267/3267 [==============================] - 5s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 160/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 161/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 162/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 163/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5913 - acc: 0.7224
Epoch 164/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 165/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5916 - acc: 0.7224
Epoch 166/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 167/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 168/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5918 - acc: 0.7224
Epoch 169/250


3267/3267 [==============================] - 5s 1ms/step - loss: 0.5914 - acc: 0.7224
Epoch 242/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 243/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 244/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 245/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5911 - acc: 0.7224
Epoch 246/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5909 - acc: 0.7224
Epoch 247/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5916 - acc: 0.7224
Epoch 248/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5912 - acc: 0.7224
Epoch 249/250
3267/3267 [==============================] - 5s 1ms/step - loss: 0.5910 - acc: 0.7224
Epoch 250/250
1633/1633 [==============================] - 6s 4ms/step
Epoch 1/250
4900/4900 [====================

4900/4900 [==============================] - 5s 943us/step - loss: 0.4238 - acc: 0.8029
Epoch 72/250
4900/4900 [==============================] - 5s 939us/step - loss: 0.4265 - acc: 0.7998
Epoch 73/250
4900/4900 [==============================] - 5s 947us/step - loss: 0.4260 - acc: 0.8002
Epoch 74/250
4900/4900 [==============================] - 5s 936us/step - loss: 0.4227 - acc: 0.8084
Epoch 75/250
4900/4900 [==============================] - 5s 945us/step - loss: 0.4245 - acc: 0.8031
Epoch 76/250
4900/4900 [==============================] - 5s 940us/step - loss: 0.4284 - acc: 0.8061
Epoch 77/250
4900/4900 [==============================] - 5s 938us/step - loss: 0.4227 - acc: 0.8035
Epoch 78/250
4900/4900 [==============================] - 5s 947us/step - loss: 0.4193 - acc: 0.8047
Epoch 79/250
4900/4900 [==============================] - 5s 938us/step - loss: 0.4233 - acc: 0.7988
Epoch 80/250
4900/4900 [==============================] - 5s 940us/step - loss: 0.4209 - acc: 0.8043
Epo

4900/4900 [==============================] - 5s 947us/step - loss: 0.4187 - acc: 0.8039
Epoch 152/250
4900/4900 [==============================] - 5s 945us/step - loss: 0.4148 - acc: 0.8078
Epoch 153/250
4900/4900 [==============================] - 5s 942us/step - loss: 0.4155 - acc: 0.8069
Epoch 154/250
4900/4900 [==============================] - 5s 944us/step - loss: 0.4181 - acc: 0.7998
Epoch 155/250
4900/4900 [==============================] - 5s 939us/step - loss: 0.4157 - acc: 0.8080
Epoch 156/250
4900/4900 [==============================] - 5s 942us/step - loss: 0.4203 - acc: 0.8049
Epoch 157/250
4900/4900 [==============================] - 5s 941us/step - loss: 0.4193 - acc: 0.8020
Epoch 158/250
4900/4900 [==============================] - 5s 942us/step - loss: 0.4162 - acc: 0.8018
Epoch 159/250
4900/4900 [==============================] - 5s 946us/step - loss: 0.4152 - acc: 0.8031
Epoch 160/250
4900/4900 [==============================] - 5s 943us/step - loss: 0.4182 - acc: 0

4900/4900 [==============================] - 5s 945us/step - loss: 0.4146 - acc: 0.8022
Epoch 232/250
4900/4900 [==============================] - 5s 941us/step - loss: 0.4149 - acc: 0.8045
Epoch 233/250
4900/4900 [==============================] - 5s 944us/step - loss: 0.4195 - acc: 0.8035
Epoch 234/250
4900/4900 [==============================] - 5s 941us/step - loss: 0.4119 - acc: 0.8071
Epoch 235/250
4900/4900 [==============================] - 5s 947us/step - loss: 0.4159 - acc: 0.8090
Epoch 236/250
4900/4900 [==============================] - 5s 941us/step - loss: 0.4130 - acc: 0.8055
Epoch 237/250
4900/4900 [==============================] - 5s 938us/step - loss: 0.4119 - acc: 0.8053
Epoch 238/250
4900/4900 [==============================] - 5s 942us/step - loss: 0.4131 - acc: 0.8102
Epoch 239/250
4900/4900 [==============================] - 5s 938us/step - loss: 0.4151 - acc: 0.8086
Epoch 240/250
4900/4900 [==============================] - 5s 939us/step - loss: 0.4146 - acc: 0

In [11]:
# summarize results
print("Best: %f using %s" % (grid_result.best_score_, grid_result.best_params_))
means = grid_result.cv_results_['mean_test_score']
stds = grid_result.cv_results_['std_test_score']
params = grid_result.cv_results_['params']
for mean, stdev, param in zip(means, stds, params):
    print("%f (%f) with: %r" % (mean, stdev, param))

Best: 0.248283 using {'epochs': 100, 'hidden_unit': 48, 'l2_weight': 0.001, 'n_layer': 4}
0.070421 (0.110061) with: {'epochs': 100, 'hidden_unit': 48, 'l2_weight': 0.001, 'n_layer': 3}
0.248283 (0.138217) with: {'epochs': 100, 'hidden_unit': 48, 'l2_weight': 0.001, 'n_layer': 4}
0.056762 (0.056762) with: {'epochs': 100, 'hidden_unit': 48, 'l2_weight': 0.01, 'n_layer': 3}
0.000000 (0.000000) with: {'epochs': 100, 'hidden_unit': 48, 'l2_weight': 0.01, 'n_layer': 4}
0.092426 (0.053107) with: {'epochs': 100, 'hidden_unit': 64, 'l2_weight': 0.001, 'n_layer': 3}
0.110435 (0.047167) with: {'epochs': 100, 'hidden_unit': 64, 'l2_weight': 0.001, 'n_layer': 4}
0.095785 (0.095785) with: {'epochs': 100, 'hidden_unit': 64, 'l2_weight': 0.01, 'n_layer': 3}
0.000000 (0.000000) with: {'epochs': 100, 'hidden_unit': 64, 'l2_weight': 0.01, 'n_layer': 4}
0.074381 (0.114191) with: {'epochs': 250, 'hidden_unit': 48, 'l2_weight': 0.001, 'n_layer': 3}
0.086612 (0.119223) with: {'epochs': 250, 'hidden_unit': 48

## Best LSTM params
best params = {'epochs': 236, 'n_layer': 3, 'hidden_unti': 64, 'l2_weight': 0}


## We have 2984 positive and 984 negtive cases. 

In [ ]:
#Make predictions for train set   
print(collections.Counter(train_Y))
print_report_for_binary_classfier(train_Y, lstm_clf.predict(train_X))

In [ ]:
#Make predictions from dev set
print(collections.Counter(val_Y))
print_report_for_binary_classfier(val_Y, lstm_clf.predict((val_X)))

In [32]:
#Make predictions for test set
print(collections.Counter(test_Y))
print_report_for_binary_classfier(test_Y, lstm_clf.predict((test_X)))

Counter({0.0: 811, 1.0: 30})
841/841 [==============================] - 6s 7ms/step
F1 score: 0.000000
precision score: 0.000000
recall score: 0.000000
accuracy score: 0.964328
matthews_corrcoef: 0.000000

Confusion matrix:
[[811   0]
 [ 30   0]]


/home/kevin/anaconda2/envs/py36/lib/python3.6/site-packages/sklearn/metrics/classification.py:1437: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 due to no predicted samples.
  'precision', 'predicted', average, warn_for)
/home/kevin/anaconda2/envs/py36/lib/python3.6/site-packages/sklearn/metrics/classification.py:1437: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 due to no predicted samples.
  'precision', 'predicted', average, warn_for)


# Save the best model

In [26]:
# from keras.models import load_model

# model.save(params.BEST_LSTM_MODEL_PATH)  # creates a HDF5 file 'my_model.h5'